packages/react/src/Heading/Heading.tsx

```diff
@@ -19,7 +19,10 @@ export type HeadingProps = {
   color?: 'inverse'
   /** The hierarchical level within the document. */
   level: 1 | 2 | 3 | 4
-  /** Uses larger or smaller text without changing its position in the heading hierarchy. */
+  /**
+   * Uses larger or smaller text without changing its position in the heading hierarchy.
+   * The value ‘level-6’ is deprecated. Use level 5 instead.
```

Comment:
We might be able to deprecate a type in a union by defining 2 union types and joining them.

Something like:

```
/** @deprecated */
type DeprecatedSize: 'level-1' | 'level-2' | 'level-3' | 'level-4' | 'level-5' | 'level-6'
type Size: 'level-1' | 'level-2' | 'level-3' | 'level-4' | 'level-5'
```

And then doing:

```
size?: Size | DeprecatedSize
```

I'm not sure this works however, and what it looks like in an IDE. But might be worth looking into.